### PR TITLE
Feature: Refactor Result and bump Dependencies

### PR DESCRIPTION
**Pull request by gaebel**
- Bumped Kotlin (to 1.6.10) and the corresponding dependencies.
- Decided to bring the Result type closer to the one of the stdlb. Added `Deprecation` annotations with `ReplaceWith` for convenience.

### CODE REVIEW

File: buildSrc/src/main/java/Libs.kt [hunk 1]
```diff
@@ -10,12 +10,12 @@ object Android {
 
 object Versions {
     const val gradle = "7.0.0"
-    const val kotlin = "1.5.31"
+    const val kotlin = "1.6.10"
     const val npmPublish = "2.1.1"
     const val definitions = "4.41.0"
-    const val coroutines = "1.5.2-native-mt"
+    const val coroutines = "1.6.0-native-mt"
     const val serialization = "1.3.0"
```
Comment (gaebel) [inline buildSrc/src/main/java/Libs.kt:17]:
```suggestion
    const val serialization = "1.3.2"
```

File: src/commonTest/kotlin/com/liftric/cognito/idp/ResultTests.kt [hunk 2]
```diff
@@ -130,7 +125,7 @@ class ResultTests {
     fun testOnResultChainingSuccess() {
         val expected = "Test"
         val result = Result.success(expected)
-            .onResult {
+            .onSuccess {
```
Comment (gaebel) [inline src/commonTest/kotlin/com/liftric/cognito/idp/ResultTests.kt:128]:
```suggestion
            .onResult {
```

**Comment by gaebel** [comment]
Going to add the missing documentation.

**Review by Ingwersaft — APPROVED** [review]
LGTM!

**Pull request merged**